app/__main__.py

```diff
@@ -3,11 +3,57 @@
 import discord
 import dotenv
 from discord import app_commands
+from discord.ui import Button, View
 
 dotenv.load_dotenv()
 TOKEN = os.environ["TOKEN"]
 
 
+class UpgradeView(View):
+    """A discord.View subclass to handle user interactions with the update screen."""
+
+    def __init__(self) -> None:
+        super().__init__(timeout=None)
+
+    @discord.ui.button(label="Upgrade CPS", style=discord.ButtonStyle.blurple, custom_id="upgradepersistent:cps")
+    async def cps_upgrade(self, interaction: discord.Interaction, button: Button) -> None:
+        """Upgrade a user's CPS."""
+        button.label = "CPS Selected"
+        if interaction.message is not None:
+            await interaction.message.edit(view=self)
```

Comment:
IMO, we shouldn't edit the source message but instead confirm in a new message like you do later. This way someone can upgrade CPS multiple times. (Similarly for other things)